dex/farm/src/base_functions.rs

```diff
@@ -153,44 +104,72 @@ pub trait BaseFunctionsModule:
     }
 
     fn merge_farm_tokens(&self) -> EsdtTokenPayment<Self::Api> {
-        let payments = self.call_value().all_esdt_transfers();
+        let mut payments = self.get_non_empty_payments();
+        let first_payment = self.pop_first_payment(&mut payments);
 
-        let attrs = self.get_default_merged_farm_token_attributes(&payments, Option::None);
-        let farm_token_id = self.farm_token().get_token_id();
-        self.burn_farm_tokens_from_payments(&payments);
+        let token_mapper = self.farm_token();
+        let mut output_attributes: FarmTokenAttributes<Self::Api> = self.get_attributes_as_part_of_fixed_supply(&first_payment, &token_mapper);
+        token_mapper.nft_burn(first_payment.token_nonce, &first_payment.amount);
 
-        self.mint_farm_tokens(farm_token_id, attrs.current_farm_amount.clone(), &attrs)
+        for payment in &payments {
+            let attributes: FarmTokenAttributes<Self::Api> = self.get_attributes_as_part_of_fixed_supply(&payment, &token_mapper);
+            output_attributes.merge_with(attributes);
+        }
```

Comment:
Refactor suggestion as to eliminate some code duplication - group these into an utility function:
`self.merge_payments(&mut output_attributes, &payments);`

Reply:
Done.